File: src/main/java/com/jss/jssbackend/modules/miscellaneous/service/VatService.java
package com.jss.jssbackend.modules.miscellaneous.service;

import java.util.List;

import com.jss.jssbackend.modules.miscellaneous.model.Vat;

public interface VatService {
    public List<Vat> getVat();

    public Vat getVat(Integer id);
}

File: client/src/app/modules/miscellaneous/services/vat.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { Vat } from '../../miscellaneous/model/Vat';

@Injectable({
  providedIn: 'root'
})
export class VatService extends AppRestService<Vat>{

  constructor(http: HttpClient) {
    super(http, "miscellaneous");
  }

  getVats() {
    return this.getListCached(new HttpParams(), "vats");
  }

  addOrUpdateVat(vat: Vat) {
    this.clearListCache(new HttpParams(), "vats");
    return this.addOrUpdate(new HttpParams(), "vat", vat, "Enregistré", "Erreur lors de l'enregistrement");
  }

}
